src/lib/swapper/swappers/ThorchainSwapper/utils/getQuote/getQuote.ts

```diff
@@ -0,0 +1,71 @@
+import type { Asset } from '@shapeshiftoss/asset-service'
+import type { AssetId } from '@shapeshiftoss/caip'
+import { adapters } from '@shapeshiftoss/caip'
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import { bn } from 'lib/bignumber/bignumber'
+import { toBaseUnit } from 'lib/math'
+import type { SwapErrorRight } from 'lib/swapper/api'
+import { makeSwapErrorRight, SwapErrorType } from 'lib/swapper/api'
+import type {
+  ThorchainSwapperDeps,
+  ThornodeQuoteResponse,
+  ThornodeQuoteResponseSuccess,
+} from 'lib/swapper/swappers/ThorchainSwapper/types'
+import { THORCHAIN_FIXED_PRECISION } from 'lib/swapper/swappers/ThorchainSwapper/utils/constants'
+
+import { thorService } from '../thorService'
+
+export const getQuote = async ({
+  sellAsset,
+  buyAssetId,
+  sellAmountCryptoBaseUnit,
+  receiveAddress,
+  affiliateBps = '0',
+  deps,
+}: {
+  sellAsset: Asset
+  buyAssetId: AssetId
+  sellAmountCryptoBaseUnit: string
+  receiveAddress: string
+  affiliateBps?: string
+  deps: ThorchainSwapperDeps
+}): Promise<Result<ThornodeQuoteResponseSuccess, SwapErrorRight>> => {
+  const buyPoolId = adapters.assetIdToPoolAssetId({ assetId: buyAssetId })
+  const sellPoolId = adapters.assetIdToPoolAssetId({ assetId: sellAsset.assetId })
+
+  const sellAmountCryptoPrecision = bn(sellAmountCryptoBaseUnit).div(
+    bn(10).pow(sellAsset.precision),
+  )
+  // All THORChain pool amounts are base 8 regardless of token precision
+  const sellAmountCryptoThorBaseUnit = bn(
+    toBaseUnit(sellAmountCryptoPrecision, THORCHAIN_FIXED_PRECISION),
+  )
+
+  const { data } = await thorService.get<ThornodeQuoteResponse>(
+    `${deps.daemonUrl}/lcd/thorchain/quote/swap?amount=${sellAmountCryptoThorBaseUnit}&from_asset=${sellPoolId}&to_asset=${buyPoolId}&destination=${receiveAddress}&affiliate_bps=${affiliateBps}&affiliate=ss`,
+  )
+
+  if ('error' in data && /not enough fee/.test(data.error)) {
+    // TODO(gomes): How much do we want to bubble the error property up?
```

Comment:
The bulk of this is the same logic as already exists in `getTradeRate.ts`, though this helper returns the entire response.